src/states/elder/mod.rs

```diff
@@ -363,8 +364,7 @@ impl Elder {
 
     fn establish_connection(&mut self, node: P2pNode) {
         self.send_direct_message(node.connection_info(), DirectMessage::ConnectionResponse);
-        self.peer_map_mut()
-            .insert(*node.public_id(), node.into_connection_info());
+        self.peer_map_mut().connect(node.connection_info().clone())
```

Comment:
Do we still need to do this? We already have the full connection info. I think it should be sufficient to call `peer_map.connect` only in `handle_connected_to`. This also applies to the other places where we do this. Similarly, we should probably only do `peer_map.disconnect` in `handle_connection_failure`.

Reply:
Agree, I've started removing the `connect` calls to see if it still works

Reply:
What about `connect` in `send_bootstrap_request`? I think we still need it there

Reply:
I'm not sure. Do we? I think it's the same thing. When we receive a response, that response will be preceded by `ConnectedTo` event from quic-p2p and we would `peer_map.connect` there.

Reply:
All the tests failed when I tried removing that one, just thought I'd check before started investigating the failures

Reply:
Hm, OK. Then maybe put it back. There might be something subtle going on in `BootstrappingPeer`. We can always revisit it later.